File: ScaleformUI_Csharp/Menus/RadialMenu/SegmentItem.cs
using ScaleformUI.Elements;

namespace ScaleformUI.Radial
{
    public class SegmentItem
    {
        private string label;
        private string description;
        private SColor color;
        private int textureHeight;
        private int textureWidth;
        private string textureName;
        private string textureDict;
        internal RadialSegment Parent;
        internal int qtty = 0;
        internal int max = 0;

        public string Label
        {
            get => label;
            set
            {
                label = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }
        public string Description
        {
            get => description;
            set
            {
                description = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }
        public string TextureDict
        {
            get => textureDict;
            set
            {
                textureDict = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }
        public string TextureName
        {
            get => textureName;
            set
            {
                textureName = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }
        public int TextureWidth
        {
            get => textureWidth;
            set
            {
                textureWidth = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }
        public int TextureHeight
        {
            get => textureHeight;
            set
            {
                textureHeight = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }
        public SColor Color
        {
            get => color;
            set
            {
                color = value;
                if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
                {
                    Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
                }
            }
        }

        /// <summary>
        /// Set the quantity of this item, if max == 0 then the quantity will be centered
        /// </summary>
        /// <param name="qtty"></param>
        /// <param name="max"></param>
        public void SetQuantity(int qtty, int max = 0)
        {
            this.qtty = qtty;
            this.max = max;
            if (Parent != null && Parent.Parent != null && Parent.Parent.Visible)
            {
                Main.radialMenu.CallFunction("UPDATE_SUBITEM", Parent.index, Parent.Items.IndexOf(this), label, description, textureName, textureDict, textureHeight, textureWidth, color, qtty, max);
            }
        }

        public SegmentItem(string label, string desc, string txd, string txn, int txwidth, int txheight, SColor color)
        {
            Label = label;
            Description = desc;
            TextureDict = txd;
            TextureName = txn;
            TextureWidth = txwidth;
            TextureHeight = txheight;
            Color = color;
        }
    }
}

File: ScaleformUI_Lua/src/Menus/RadialMenu/SegmentItem.lua
SegmentItem = setmetatable({}, SegmentItem)
SegmentItem.__index = SegmentItem
SegmentItem.__call = function()
    return "SegmentItem"
end

---@class SegmentItem
---@field public label string
---@field public description string
---@field public textureDict string
---@field public textureName string
---@field public textureWidth number
---@field public textureHeight number
---@field public color SColor
---@field public qtty number
---@field public max number
---@field public Parent SegmentItem

---New
---@param _label string
---@param _desc? string
---@param _txd? string
---@param _txn? string
---@param _txwidth? number
---@param _txheight? number
---@param _color? SColor
---@return SegmentItem
function SegmentItem.New(_label, _desc, _txd, _txn, _txwidth, _txheight, _color)
    local _it = {
        label = _label or "",
        description = _desc or "",
        textureDict = _txd or "",
        textureName = _txn or "",
        textureWidth = _txwidth or 0,
        textureHeight = _txheight or 0,
        color = _color or SColor.HUD_Freemode,
        qtty = 0,
        max = 0,
        Parent = nil
    }
    return setmetatable(_it, SegmentItem)
end

function SegmentItem:Label(lbl)
    if lbl ~= nil then
        self.label = lbl
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.label
    end
end

function SegmentItem:Description(desc)
    if desc ~= nil then
        self.description = desc
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.description
    end
end

function SegmentItem:TextureDict(txd)
    if txd ~= nil then
        self.textureDict = txd
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.textureDict
    end
end

function SegmentItem:TextureName(txn)
    if txn ~= nil then
        self.textureName = txn
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.textureName
    end
end

function SegmentItem:TextureWidth(width)
    if width ~= nil then
        self.textureWidth = width
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.textureWidth
    end
end

function SegmentItem:TextureHeight(height)
    if height ~= nil then
        self.textureHeight = height
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.textureHeight
    end
end

function SegmentItem:Color(color)
    if color ~= nil then
        self.color = color
        if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
            ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
        end
    else
        return self.color
    end
end

function SegmentItem:SetQuantity(qtty, max)
    self.qtty = qtty
    self.max = max or 0
    if self.Parent ~= nil and self.Parent.Parent ~= nil and self.Parent.Parent:Visible() then
        ScaleformUI.Scaleforms._radialMenu:CallFunction("UPDATE_SUBITEM", self.Parent.index - 1, IndexOf(self.Parent.Items, self) - 1, self.label, self.description, self.textureName, self.textureDict, self.textureHeight, self.textureWidth, self.color, self.qtty, self.max);
    end
end
